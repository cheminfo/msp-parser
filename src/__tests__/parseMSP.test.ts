import { readFileSync } from 'fs';
import { join } from 'path';

import { parseMSP } from '../index';

test('parseMSP', () => {
  const blob = readFileSync(join(__dirname, 'massBank.msp'));
  const results = parseMSP(blob);
  expect(results).toHaveLength(2);
  // @ts-expect-error It is a test we don't care
  expect(Object.keys(results[0].meta)).toStrictEqual([
    'Name',
    'Synon',
    'DB#',
    'InChIKey',
    'InChI',
    'SMILES',
    'Precursor_type',
    'Spectrum_type',
    'PrecursorMZ',
    'Instrument_type',
    'Instrument',
    'Ion_mode',
    'Collision_energy',
    'Formula',
    'MW',
    'ExactMass',
    'Comments',
    'Splash',
    'Num Peaks',
  ]);
  expect(results[0].variables).toStrictEqual({
    x: {
      data: [
        91.0542, 125.0233, 154.0499, 155.0577, 185.0961, 200.107, 229.0859,
        246.1125,
      ],
      label: 'm/z',
      units: 'u',
    },
    y: {
      data: [245, 999, 80, 355, 349, 45, 142, 734],
      label: 'Relative intensity',
      units: '%',
    },
  });

  expect(results).toMatchSnapshot();
});
