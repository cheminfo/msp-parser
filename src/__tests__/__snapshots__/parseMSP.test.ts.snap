// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseMSP 1`] = `
[
  {
    "meta": {
      "Collision_energy": "30(NCE)",
      "Comments": "Parent=288.1225",
      "DB#": "MSBNK-AAFC-AC000854",
      "ExactMass": "287.11575",
      "Formula": "C16H17NO4",
      "InChI": "InChI=1S/C16H17NO4/c1-11(18)17-14(8-12-6-4-3-5-7-12)15-9-13(20-2)10-16(19)21-15/h3-7,9-10,14H,8H2,1-2H3,(H,17,18)/t14-/m0/s1",
      "InChIKey": "VFMQMACUYWGDOJ-AWEZNQCLSA-N",
      "Instrument": "Q-Exactive Orbitrap Thermo Scientific",
      "Instrument_type": "LC-ESI-ITFT",
      "Ion_mode": "POSITIVE",
      "MW": "287",
      "Name": "Pyrophen",
      "Num Peaks": "8",
      "PrecursorMZ": "288.1225",
      "Precursor_type": "[M+H]+",
      "SMILES": "CC(=O)N[C@@H](CC1=CC=CC=C1)C2=CC(=CC(=O)O2)OC",
      "Spectrum_type": "MS2",
      "Splash": "splash10-004j-1940000000-b1bd14eb30f6afd2739e",
      "Synon": "N-[(1S)-1-(4-methoxy-6-oxopyran-2-yl)-2-phenylethyl]acetamide",
    },
    "variables": {
      "x": {
        "data": [
          91.0542,
          125.0233,
          154.0499,
          155.0577,
          185.0961,
          200.107,
          229.0859,
          246.1125,
        ],
        "label": "m/z",
        "units": "u",
      },
      "y": {
        "data": [
          245,
          999,
          80,
          355,
          349,
          45,
          142,
          734,
        ],
        "label": "Relative intensity",
        "units": "%",
      },
    },
  },
  {
    "meta": {
      "Collision_energy": "50(NCE)",
      "Comments": "Parent=319.1535",
      "DB#": "MSBNK-AAFC-AC000292",
      "ExactMass": "318.14672",
      "Formula": "C18H22O5",
      "InChI": "InChI=1S/C18H22O5/c1-12-6-5-9-14(19)8-4-2-3-7-13-10-15(20)11-16(21)17(13)18(22)23-12/h3,7,10-12,20-21H,2,4-6,8-9H2,1H3/b7-3+/t12-/m0/s1",
      "InChIKey": "MBMQEIFVQACCCH-QBODLPLBSA-N",
      "Instrument": "Q-Exactive Orbitrap Thermo Scientific",
      "Instrument_type": "LC-ESI-ITFT",
      "Ion_mode": "POSITIVE",
      "MW": "318",
      "Name": "Zearalenone",
      "Num Peaks": "72",
      "PrecursorMZ": "319.1535",
      "Precursor_type": "[M+H]+",
      "SMILES": "C[C@H]1CCCC(=O)CCC/C=C/C2=CC(=CC(=C2C(=O)O1)O)O",
      "Spectrum_type": "MS2",
      "Splash": "splash10-000i-2930000000-5f23d1f8df586bd436be",
      "Synon": "(2E,11S)-15,17-dihydroxy-11-methyl-12-oxabicyclo[12.4.0]octadeca-1(14),2,15,17-tetraene-7,13-dione",
    },
    "variables": {
      "x": {
        "data": [
          55.0185,
          67.0542,
          68.9971,
          69.0699,
          71.0491,
          79.0542,
          81.0699,
          83.0491,
          85.0648,
          91.0542,
          93.0699,
          95.0491,
          95.0855,
          97.0648,
          105.0699,
          107.0855,
          115.0542,
          117.0699,
          119.0855,
          123.0441,
          125.0961,
          129.0699,
          131.0855,
          133.0648,
          141.0699,
          145.0648,
          147.0441,
          147.0804,
          149.0597,
          157.0648,
          159.0804,
          160.0519,
          161.0597,
          163.039,
          163.0754,
          165.0182,
          169.0648,
          173.0597,
          175.039,
          175.0754,
          177.0546,
          177.091,
          185.0597,
          187.0754,
          189.0546,
          197.0597,
          199.0754,
          201.091,
          203.0703,
          205.0495,
          205.0859,
          211.0754,
          213.0546,
          213.091,
          214.0624,
          215.0703,
          215.1067,
          217.0859,
          223.0754,
          225.091,
          226.0624,
          227.0703,
          228.0781,
          229.0859,
          231.0652,
          241.0859,
          241.1223,
          243.1016,
          250.0988,
          255.138,
          265.1223,
          283.1329,
        ],
        "label": "m/z",
        "units": "u",
      },
      "y": {
        "data": [
          114,
          76,
          40,
          838,
          32,
          62,
          55,
          40,
          122,
          94,
          51,
          32,
          56,
          237,
          79,
          47,
          53,
          81,
          59,
          103,
          45,
          111,
          133,
          40,
          207,
          39,
          70,
          64,
          68,
          453,
          78,
          68,
          314,
          60,
          56,
          130,
          119,
          98,
          214,
          356,
          128,
          108,
          999,
          895,
          481,
          63,
          106,
          77,
          802,
          75,
          122,
          42,
          159,
          123,
          37,
          112,
          43,
          41,
          54,
          73,
          33,
          121,
          36,
          91,
          55,
          89,
          36,
          41,
          46,
          39,
          86,
          44,
        ],
        "label": "Relative intensity",
        "units": "%",
      },
    },
  },
]
`;
